Address Issues #1 and #3.

Flameeyes opened the pull request:
This includes the usage of pkgconfig, and some fixes to the Cython code to make it more stable.

Flameeyes committed:
Use pkgconfig module to find libiscsi when building the extension.

This allows building on Fedora and other distributions that might put the
library in different paths, as well as reporting a proper error if a
too-old version of the library is installed.

Flameeyes commented:
Markus, if you can test this pull request, I think it addresses the problem.

If it works, mark the request as approved, and Mergify will take care of the rest ;)

Flameeyes committed:
Remove has_datain/has_dataout attributes.

These were leftovers from before I implemented `Context.command()` to set
the datain/dataout buffers.

Flameeyes committed:
Use a const memoryview to accept the cdb, rather than bytes.

This accepts both bytes and bytearray objects as input.

See recommendation at https://github.com/cython/cython/issues/3488.

rosjat commented:
i will do when we figured the stuff with the iscsi too then we can replace the old sgio and iscsi modules

Flameeyes committed:
Validate `iscsi.Task()` `cdb` argument.

This just makes the error more obvious, rather than stack tracing deep into
the wrapper.

Flameeyes committed:
Correct data in/out buffer handling, they can all fail.

While `iscsi.h` does not document the return value, the
`scsi_task_add_data_*_buffer`interfaces return an error if the length is
zero, rather than ignoring it altogether.
